rustls/src/msgs/handshake.rs

```diff
@@ -359,16 +358,48 @@ impl ConvertProtocolNameList for Vec<ProtocolName> {
             .map(|proto| proto.as_ref())
             .collect::<Vec<&[u8]>>()
     }
+}
+
+/// RFC7301 encodes a single protocol name as `Vec<ProtocolName>`
+#[derive(Clone, Debug)]
+pub struct SingleProtocolName(ProtocolName);
+
+impl SingleProtocolName {
+    pub(crate) fn new(name: impl Into<Vec<u8>>) -> Self {
```

Comment:
Given that this is internal, I'd externalize the conversion to `Vec<u8>` (by doing away with the `impl Into<_>` here).